src/main/java/uk/gov/pay/publicauth/util/ApplicationStartupDependentResourceChecker.java

```diff
@@ -3,42 +3,30 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import java.sql.SQLException;
+import java.time.Duration;
+import java.util.function.Consumer;
 
-public class ApplicationStartupDependentResourceChecker {
+class ApplicationStartupDependentResourceChecker {
 
     private static final int PROGRESSIVE_SECONDS_TO_WAIT = 5;
     private static final Logger logger = LoggerFactory.getLogger(ApplicationStartupDependentResourceChecker.class);
 
     private final ApplicationStartupDependentResource applicationStartupDependentResource;
+    private final Consumer<Duration> waiter;
 
-    public ApplicationStartupDependentResourceChecker(ApplicationStartupDependentResource applicationStartupDependentResource) {
+    ApplicationStartupDependentResourceChecker(ApplicationStartupDependentResource applicationStartupDependentResource, Consumer<Duration> waiter) {
         this.applicationStartupDependentResource = applicationStartupDependentResource;
+        this.waiter = waiter;
     }
 
-    public void checkAndWaitForResources() {
-        waitingForDatabaseConnectivity();
-    }
-
-    private void waitingForDatabaseConnectivity() {
-        logger.info("Checking for database availability >>>");
-
+    void checkAndWaitForResource() {
         long timeToWait = 0;
-        while(!isDatabaseAvailable()) {
+        while(!applicationStartupDependentResource.isAvailable()) {
             timeToWait += PROGRESSIVE_SECONDS_TO_WAIT;
-            logger.info("Waiting for {} seconds till the database is available ...", timeToWait);
-            applicationStartupDependentResource.sleep(timeToWait * 1000);
+            logger.info("Waiting for {} seconds until {} is available ...", timeToWait, applicationStartupDependentResource);
+            waiter.accept(Duration.ofSeconds(timeToWait));
         }
         logger.info("Database available.");
```

Comment:
I guess, if we want applicationStartupDependentResource to be generic (an interface) with different implementations, we should avoid logging database here, because we might be waiting on other things

Reply:
Ah, yes, good point

Reply:
Fix pushed